HW_5/HW05/HW05.Task4/Program.cs

```diff
@@ -0,0 +1,148 @@
+
+                        // step back one iteration, because the string has become shorter
+                        // (so as not to skip checking the next value)
+                        j--;
+                    }
+                    // checking "if the character is a space, then"
+                    else if (String.Equals(necessaryAdressesArray[i][j], ' '))
+                    {
+                        // trim the spaces at the beginning and end of the line
+                        necessaryAdressesArray[i] = necessaryAdressesArray[i].Trim();
+
+                        // exit the cycle "for"
+                        break;
+                    }
+                }
+                // add the resulting house number to the corresponding list
+                houseNumbers.Add(houseNumber);
+
+                // reset the value of the variable "houseNumber"
+                houseNumber = null;
+            }
+            // create a StringBuilder object that will contain the entered zip code, the corresponding addresses and house numbers
+            StringBuilder zipCodeAddresses = new StringBuilder();
+
+            // add the entered zip code and a colon to the object "zipCodeAddresses"
+            zipCodeAddresses.Append(zipCode + ':');
+
+            // go through each element of the array of necessary addresses and add them to the object "zipCodeAddresses", separated
+            // by commas
+            for (int i = 0; i < necessaryAdressesArray.Length; i++)
+            {
+                zipCodeAddresses.Append(necessaryAdressesArray[i] + ',');
+            }
+
+            // convert the list of addresses home to an array
+            string[] houseNumbersToZipCodeArray = houseNumbers.ToArray();
+
+            // enter the constant for the length of the zip code (including the colon)
+            const int ZipCodeLength = 9;
+
+            // check the length of the "zipCodeAddresses" object with the entered constant
+            if (zipCodeAddresses.Length == ZipCodeLength)
+            {
+                // add a slash if the length matches
+                zipCodeAddresses.Append('/');
+            }
+            else
+            {
+                // remove the last extra comma in the string
+                zipCodeAddresses.Remove(zipCodeAddresses.Length - 1, 1);
+
+                // add a slash
+                zipCodeAddresses.Append('/');
+
+                // add all house numbers separated by commas to the address line
+                for (int i = 0; i < houseNumbersToZipCodeArray.Length; i++)
+                {
+                    zipCodeAddresses.Append(houseNumbersToZipCodeArray[i] + ',');
+                }
+                // remove the last extra comma in the string
+                zipCodeAddresses.Remove(zipCodeAddresses.Length - 1, 1);
+            }
+            return zipCodeAddresses.ToString();
+        }
+    }
+}
```

Comment:
it works, but I would reduce some extra steps (some loops). also you can avoid using List there. Use StringBuilder instead.